scripts/imgtool/main.py

```diff
@@ -277,7 +279,21 @@ def sign(key, align, version, header_size, pad_header, slot_size, pad,
             # FIXME
             raise click.UsageError("Signing and encryption must use the same "
                                    "type of key")
-    img.create(key, enckey, dependencies)
+
+    # Get list of custom protected TLVs from the command-line
+    custom_tlv = {}
+    for tlv in cptlv:
+        tag = int(tlv[0], 16) if tlv[0].startswith('0x') else int(tlv[0])
+        if tag in custom_tlv:
+            raise ValueError('Custom TLV %s already exists.' % hex(tag))
+        elif tag in image.TLV_VALUES.values():
+            raise ValueError('Custom TLV %s conflicts with predefined TLV.' % hex(tag))
+        else:
+            length = tlv[1]
+            value = tlv[2]
```

Comment:
I would convert both to `int` already: `length, value = int(tlv[1], 0), int(tlv[2], 0)`